app/routes/circulars/index.tsx

```diff
@@ -151,12 +184,20 @@ export default function () {
   const allItems = [...(newItem ? [newItem] : []), ...(items || [])]
 
   const [searchParams] = useSearchParams()
+  console.log('default function search params', searchParams.toString())
   const query = searchParams.get('query') ?? undefined
+  const startDate = searchParams.get('startDate') ?? undefined
+  const endDate = searchParams.get('endDate') ?? undefined
+
   let searchParamsString = searchParams.toString()
   if (searchParamsString) searchParamsString = `?${searchParamsString}`
+  const combinedQuery = { query, page: page, startDate, endDate }
+  console.log('default function combinedQuery', combinedQuery)
 
-  const [input, setInput] = useState(query)
-  const clean = input === query
+  const [input, setInput] = useState(combinedQuery)
+  const clean = input === combinedQuery
```

Comment:
My advice: don't create states that are composite objects. Have a separate state variable for each input field, like this:

```ts
const [inputQuery, setInputQuery] = useState(query)
const [inputDateGte, setInputDateGte] = useState(gte)
const [inputDateLte, setInputDateLte] = useState(lte)
const clean = inputQuery === query && inputDateGte === gte && inputDateLte === lte
...
```

The reason is that there are [many subtleties around properly updating React states that are objects](https://react.dev/learn/updating-objects-in-state).

Reply:
this actually fixed the issue with the loading not working I think